llama-index-integrations/vector_stores/llama-index-vector-stores-azurecosmosnosql/llama_index/vector_stores/azurecosmosnosql/base.py

```diff
@@ -0,0 +1,246 @@
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.schema import BaseNode, MetadataMode, TextNode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from llama_index.core.vector_stores.utils import (
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AzureCosmosDBNoSqlVectorSearch(BasePydanticVectorStore):
+    """Azure CosmosDB NoSQL vCore Vector Store.
+
+    To use, you should have both:
+    -the ``azure-cosmos`` python package installed
+    -from llama_index.vector_stores.azurecosmosnosql import AzureCosmosDBNoSqlVectorSearch
+    """
+    stores_text: bool = True
+    flat_metadata: bool = True
+
+    _cosmos_client: Any = PrivateAttr()
+    _database_name: Any = PrivateAttr()
+    _container_name: Any = PrivateAttr()
+    _embedding_key: Any = PrivateAttr()
+    _vector_embedding_policy: Any = PrivateAttr()
+    _indexing_policy: Any = PrivateAttr()
+    _cosmos_container_properties: Any = PrivateAttr()
+    _cosmos_database_properties: Any = PrivateAttr()
+    _create_container: Any = PrivateAttr()
+    _database: Any = PrivateAttr()
+    _container: Any = PrivateAttr()
+    _id_key: Any = PrivateAttr()
+    _text_key: Any = PrivateAttr()
+    _metadata_key: Any = PrivateAttr()
+
+    def __init__(
+            self,
+            cosmos_client: Optional[Any] = None,
+            vector_embedding_policy: Optional[Dict[str, Any]] = None,
+            indexing_policy: Optional[Dict[str, Any]] = None,
+            cosmos_container_properties: Optional[Dict[str, Any]] = None,
+            cosmos_database_properties: Optional[Dict[str, Any]] = None,
+            database_name: str = "vectorSearchDB",
+            container_name: str = "vectorSearchContainer",
+            create_container: bool = True,
+            id_key: str = "id",
+            text_key: str = "text",
+            metadata_key: str = "metadata",
+            **kwargs: Any,
+    ) -> None:
+        """Initialize the vector store.
+
+        Args:
+            cosmos_client: Client used to connect to azure cosmosdb no sql account.
+            database_name: Name of the database to be created.
+            container_name: Name of the container to be created.
+            embedding: Text embedding model to use.
+            vector_embedding_policy: Vector Embedding Policy for the container.
+            indexing_policy: Indexing Policy for the container.
+            cosmos_container_properties: Container Properties for the container.
+            cosmos_database_properties: Database Properties for the container.
+        """
+        super().__init__()
+
+        if cosmos_client is not None:
+            self._cosmos_client = cast(CosmosClient, cosmos_client)
+
+        if create_container:
+            if (
+                    indexing_policy["vectorIndexes"] is None
```

Comment:
from the signature I see `indexing_policy` might be `None` and this would cause a `KeyError`

Reply:
Updated the signature, indexing policy cannot be None.